### PR TITLE
Load last channel on startup

### PR DESCRIPTION
**Pull request by nikolaiwarner**
This adds a loading screen to automatically start up your last cabal and put you into the last channel you were in on app startup.

### CODE REVIEW

**Comment by nikolaiwarner** [comment]
@staltz How does this look?

Should I always wait for a review or is it cool if I merge my own PRs? ... just to keep things moving faster

File: frontend/screens/ChannelsList.js [hunk 1]
```diff
@@ -85,16 +87,25 @@ export default class HomeScreen extends React.Component {
...
   }
 
-  updateKey(key) {
+  async updateKey(key) {
     this.props.navigation.setParams({key, nick: this.state.nick});
     this.setState(prev => ({...prev, key}));
+
+    if (this._initialLoad) {
+      this._initialLoad = false
+      const channel = (await AsyncStorage.getItem('favorite-channel'))
```
Comment (soyuka) [inline frontend/screens/ChannelsList.js]: parenthesis not needed arount `await ...`

File: frontend/screens/SplashScreen.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    this.state = {
+      logoAnimation: new Animated.Value(0)
+    }
+  }
+
+  static navigationOptions = ({navigation}) => {
+    return {
+      header: null
+    }
+  }
```
Comment (soyuka) [inline frontend/screens/SplashScreen.js]: can be simplified `({navigation}) => ({header: null})`

File: frontend/screens/SplashScreen.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      backend.channel.removeListener('message', this.backendListener)
+    }
+  }
+
+  resumeLastCabal = ({key, nick, channel}) => {
+    backend.start('main.js')
+    this.backendListener = (raw) => {
+      const msg = JSON.parse(raw)
+      if (msg.type === 'ready') {
+        console.log('READY!', msg)
```
Comment (soyuka) [inline frontend/screens/SplashScreen.js]: should we keep this? should we log every event instead in debug-mode?

**Comment by soyuka** [comment]
> Should I always wait for a review or is it cool if I merge my own PRs? ... just to keep things moving faster

I'd say always PR it's good to have second looks on the code. Only my opinion though :p.

**Comment by nikolaiwarner** [comment]
Yep! I agree on PR reviews.... just so eager :D Thanks!!

File: frontend/screens/SplashScreen.js (hunk 3 above)
Comment (soyuka) [reply frontend/screens/SplashScreen.js]: The log question remains though :D.

**Pull request merged**